tutorials/terraform-asm-upgrade.md

```diff
@@ -0,0 +1,553 @@
+        istiod-asm-196-2-8544879bbd-bfnd6                 1/1     Running   0          8m20s   asm-196-2
+        istiod-asm-196-2-8544879bbd-gp25n                 1/1     Running   0          8m20s   asm-196-2
+
+1.  Get the upgraded Anthos Service Mesh revision label:
+
+        export ISTIOD_UPGRADED_POD=$(kubectl --context=${CLUSTER_1_CTX} get pod -n istio-system -l app=istiod | grep asm-110 | awk 'NR==1 {print $1}')
+        export ASM_REV_UPGRADE=$(kubectl --context=${CLUSTER_1_CTX} -n istio-system get pod ${ISTIOD_UPGRADED_POD} \
+          -o jsonpath='{.metadata.labels.istio\.io/rev}')
+        echo $ASM_REV_UPGRADE
+
+1.  Switch the `istio-ingressgateway` to the new revision:
+
+        kubectl --context=${CLUSTER_1_CTX} patch service -n istio-system istio-ingressgateway \
+          --type='json' -p="[{"op": "replace", "path": "/spec/selector/service.istio.io~1canonical-revision", "value": "${ASM_REV_UPGRADE}"}]"
+
+1.  Add the revision label to the `online-boutique` namespace and remove the `istio-injection` label (if it exists):
+
+        kubectl --context=${CLUSTER_1_CTX} label namespace online-boutique istio.io/rev=${ASM_REV_UPGRADE} istio-injection- --overwrite
+
+   You can ignore `istio-injection not found` in the output. If you see that message, it means that the namespace didn't previously have the
+   `istio-injection` label.
+
+1.  Restart the Pods to trigger re-injection:
+
+        kubectl --context=${CLUSTER_1_CTX} rollout restart deployment -n online-boutique
+
+1.  Wait until all Deployments are restarted:
+
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment adservice 
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment checkoutservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment currencyservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment emailservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment frontend
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment paymentservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment productcatalogservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment shippingservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment cartservice
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment loadgenerator
+        kubectl --context=${CLUSTER_1_CTX} -n online-boutique rollout status -w --timeout=5m deployment recommendationservice
+
+    The output is similar to the following:
```

Comment:
This output section is useful. Include it in the other two documents?

Reply:
Added output to the other docs